src/Mjolnir.ts

```diff
@@ -638,17 +638,18 @@ export class Mjolnir {
         }
     }
 
-    public async syncListForRoom(roomId: string) {
-        let updated = false;
-        for (const list of this.banLists) {
-            if (list.roomId !== roomId) continue;
-            await list.updateList();
-            updated = true;
-        }
-        if (!updated) return;
+    /**
+     * Pulls any changes to the rules in room acting as a policy list and updates all protected rooms
+     * with those changes. Does not fail if there are errors updating the room, these are reported to the management room.
+     * @param roomId The room with a ban/policy list to update and apply to all protected rooms.
+     * @returns When all of the protected rooms have been updated.
+     */
+    public async syncListForRoom(roomId: string): Promise<void> {
```

Comment:
It really is doing two things. One is keeping the BanList model up to date which is a thing because it is used to check users on the fly as they join rooms (and probably more). The other is keeping the user bans and server ACLs in all of the protected rules synchronized with this policy list, but in order to do this correctly it has to use all of the watched policy lists (because without knowledge of the other lists it would not be able to respond to unbanning users and servers without removing bans that may have been there because of other policy lists). `syncWithPolicyRoom` is probably a better name?

Reply:
Probably better, yeah.